services/src/main/java/org/keycloak/services/HttpRequestImpl.java

```diff
@@ -61,8 +65,19 @@ public MultivaluedMap<String, String> getDecodedFormParameters() {
         }
         MediaType mediaType = getHttpHeaders().getMediaType();
         if (mediaType == null || !mediaType.isCompatible(MediaType.valueOf("application/x-www-form-urlencoded"))) {
-            return new MultivaluedHashMap<>();
+            throw new BadRequestException("MediaType is not compatible");
```

Comment:
What do you think about adding the incorrect media type to do a proper message? Something like ` "MediaType " + mediaType + " is not application/x-www-form-urlencoded compatible"`.